libr/cons/grep.c

```diff
@@ -278,6 +278,40 @@ R_API void r_cons_grep(const char *str) {
 	}
 }
 
+R_API void r_cons_grep_parsecmd(char *cmd, const char *quotestr) {
+	char *ptr;
+	int len = strlen (cmd);
+
+	//  search and modify '~str~?' into '~?str'
+	if (len > 4 && r_str_endswith (cmd, "~?") && cmd[len - 3] != '\\') {
```

Comment:
Markdown at github.com messes part of your commentary.
Just to be clear, did you mean that 

    ~str1~str2~?

should do the same as

    ~?str1str2

I.e. r2 need to handle every unescaped '~' from command string, and build one grep expression from multiple intgrep subcommands?